lib/rspec/support/differ.rb

```diff
@@ -56,6 +58,28 @@ def diff_as_string(actual, expected)
       end
       # rubocop:enable Metrics/MethodLength
 
+      def diff_hashes_as_object(actual, expected)
+        if defined?(RSpec::Mocks::ArgumentMatchers::AnyArgMatcher)
```

Comment:
We only call this from all_hashes?, and it already has this check

Potentially, this method can be mistakenly called from somewhere else causing errors for those not using rspec-mocks, but i’m less worried about this at this point.

Reply:
Yes, it is currently called only under `if all_hashes?(actual, expected)` condition, **and** `all_hashes?` is **also** checking if the user is opting out `rspec-mocks`. This check in line 62 is redundant if we assume this method *will only be called inside `all_hashes?` condition* . But it is there in case someone mistakenly call the method from somewhere else, so the program won't break if `rspec-mocks` is opted-out.

If someone opts-out `rspec-mock`, `diff_hashes_as_object` should have the exact same behavior as `diff_as_object`. That's the purpose of this redundant check.

Personally, I like redundancy. But I understand sometimes it may be overkill or unnecessary... maybe defining `diff_hashes_as_object` as private is good enough? I don't have a strong opinion here.